File: project/android/apps/MnnLlmApp/app/src/main/java/com/alibaba/mnnllm/android/utils/DeviceUtils.java
// Created by ruoyi.sjd on 2025/2/10.
// Copyright (c) 2024 Alibaba Group Holding Limited All rights reserved.

package com.alibaba.mnnllm.android.utils;
import android.os.Build;
public class DeviceUtils {

    public static String getDeviceInfo() {
        return "DeviceInfo: " + android.os.Build.MANUFACTURER + " "
                + android.os.Build.MODEL + " " + android.os.Build.VERSION.RELEASE +
                "\nSdkInt:" + Build.VERSION.SDK_INT;
    }
}

File: apps/Android/MnnTaoAvatar/app/src/main/java/com/taobao/meta/avatar/utils/DeviceUtils.kt
// Created by ruoyi.sjd on 2025/3/27.
// Copyright (c) 2024 Alibaba Group Holding Limited All rights reserved.

package com.taobao.meta.avatar.utils

import android.content.res.Configuration
import android.os.Build
import com.alibaba.mls.api.ApplicationProvider
import java.util.Locale

object DeviceUtils {
    val deviceInfo: String
        get() = ("DeviceInfo: " + Build.MANUFACTURER + " "
                + Build.MODEL + " " + Build.VERSION.RELEASE +
                "\nSdkInt:" + Build.VERSION.SDK_INT)

    val isChinese: Boolean
        get() {
            val config: Configuration = ApplicationProvider.get().resources.configuration
            val locale: Locale = config.locales.get(0)
            val language: String = locale.language
            val country: String = locale.country
            if (language == "zh" && country == "CN") {
                return true
            } else {
                return false
            }
        }
}